ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,449 @@
+    }
+
+    // MARK: - Animations
+
+    private func completePan(with velocity: CGFloat) {
+        var targetState: BottomSheetExpansionState
+        if abs(velocity) < Constants.directionOverrideVelocityThreshold {
+            // Velocity too low, snap to the closest offset
+            targetState =
+                abs(collapsedOffsetFromBottom - currentOffsetFromBottom) < abs(expandedOffsetFromBottom - currentOffsetFromBottom)
+                ? .collapsed
+                : .expanded
+        } else {
+            // Velocity high enough, animate to the offset we're swiping towards
+            targetState = velocity > 0 ? .collapsed : .expanded
+        }
+        move(to: targetState, velocity: velocity)
+    }
+
+    private func move(to targetExpansionState: BottomSheetExpansionState, animated: Bool = true, velocity: CGFloat = 0.0) {
+        if animated {
+            animate(to: targetExpansionState, velocity: velocity)
+        } else {
+            bottomSheetOffsetConstraint.constant = -(targetExpansionState == .expanded ? expandedOffsetFromBottom : collapsedOffsetFromBottom)
+            handleCompletedStateChange(to: targetExpansionState)
+        }
+    }
+
+    private func animate(to targetExpansionState: BottomSheetExpansionState, velocity: CGFloat = 0.0) {
+        let targetOffsetFromBottom = targetExpansionState == .expanded ? expandedOffsetFromBottom : collapsedOffsetFromBottom
+        let distanceToGo = abs(currentOffsetFromBottom - targetOffsetFromBottom)
+        let springVelocity = min(abs(velocity / distanceToGo), Constants.Spring.maxInitialVelocity)
+        let damping: CGFloat = abs(velocity) > Constants.Spring.flickVelocityThreshold
+            ? Constants.Spring.oscillatingDampingRatio
+            : Constants.Spring.defaultDampingRatio
+
+        let springParams = UISpringTimingParameters(dampingRatio: damping, initialVelocity: CGVector(dx: 0.0, dy: springVelocity))
+        translationAnimator = UIViewPropertyAnimator(duration: Constants.Spring.animationDuration, timingParameters: springParams)
+
+        view.layoutIfNeeded()
+        bottomSheetOffsetConstraint.constant = -targetOffsetFromBottom
+        translationAnimator?.addAnimations {
+            self.view.layoutIfNeeded()
+        }
+
+        translationAnimator?.addCompletion({ finalPosition in
+            if finalPosition == .end {
+                self.handleCompletedStateChange(to: targetExpansionState)
+            }
+        })
+        translationAnimator?.startAnimation()
+    }
+
+    private func handleCompletedStateChange(to targetExpansionState: BottomSheetExpansionState) {
+        switch targetExpansionState {
+        case .expanded:
+            self.delegate?.bottomSheetControllerDidExpand?(self)
+        case .collapsed:
+            self.delegate?.bottomSheetControllerDidCollapse?(self)
+        }
+        updateResizingHandleViewAccessibility()
```

Comment:
isn't this wayyy to late? that mean when you first set up the bottomsheet the handle will just say "button". only after user interacts with it at least once, it will have the right accessibility label

Reply:
Good catch, fixed.